src/plugins/discourse/weights.js

```diff
@@ -63,11 +65,26 @@ export const weightsConfigParser: C.Parser<WeightsConfig> = C.fmap(
   upgrade
 );
 
-export function likeWeight(user: ?User): NodeWeight {
+export function likeWeight(
+  {weights = {}}: DiscourseConfig,
+  user: ?User,
+  category: ?DiscourseCategoryId,
+  tags: ?$ReadOnlyArray<Tag>
+): NodeWeight {
   if (user == null) {
     return 0;
   }
-  return _trustLevelWeight(user);
+  if (category == null) {
+    category = -Infinity;
+  }
+  if (tags == null) {
+    tags = [];
+  }
+  return (
+    _trustLevelWeight(user) *
+    _categoryWeight(category, weights) *
+    _weightFromTags(tags, weights)
```

Comment:
I recommend switching to the approach we looked at while pair programming, like:

```js
const categoryMult = categoryId == null ? 1 : _categoryWeight(categoryId, weights);
const tagMult = ...
```